### ISSUE TITLE
Torrent Details > Files causing interface lockups with large torrents

### ISSUE THREAD

**Issue by ferdnyc** [BiglySoftware/BiglyBT#1217]
Once again I've managed to trigger a complete GUI freeze in BiglyBT, when viewing the Files list of a large torrent (literally thousands of files), when I moved too fast and attempted an action before the previous one was completed.

However, this time I thought to attach `jconsole` to the still-running (and still active, but completely unresponsive) process, and collected some data.

Whenever this happens, even though the "Detect Deadlock" button in the Thread tab never reports anything awry, I notice that the SWT Thread is always BLOCKED on a GetOffSWT thread, and there's _another_ GetOffSWT thread that's also BLOCKED on that same first GetOffSWT thread... which is waiting on a lock that will, apparently, never be released.

The three threads' stack traces:

<details>
<summary>SWT Thread</summary>

```
Name: SWT Thread
State: BLOCKED on java.lang.Object@5886ee1b owned by: GetOffSWT 1014
Total blocked: 1,771  Total waited: 1,170

Stack trace: 
com.biglybt.ui.swt.views.table.painted.TableViewPainted.swt_calculateClientArea(TableViewPainted.java:2417)
com.biglybt.ui.swt.views.table.painted.TableViewPainted.swt_paintComposite(TableViewPainted.java:1593)
com.biglybt.ui.swt.views.table.painted.TableViewPainted$14.paintControl(TableViewPainted.java:1376)
org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:234)
org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
org.eclipse.swt.widgets.Display.sendEvent(Display.java:5783)
org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1411)
org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1437)
org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1420)
org.eclipse.swt.widgets.Control.gtk_draw(Control.java:3863)
org.eclipse.swt.widgets.Composite.gtk_draw(Composite.java:455)
org.eclipse.swt.widgets.Canvas.gtk_draw(Canvas.java:181)
org.eclipse.swt.widgets.Widget.windowProc(Widget.java:2177)
org.eclipse.swt.widgets.Control.windowProc(Control.java:6764)
org.eclipse.swt.widgets.Display.windowProc(Display.java:6035)
org.eclipse.swt.internal.gtk.GTK._gtk_main_do_event(Native Method)
org.eclipse.swt.internal.gtk.GTK.gtk_main_do_event(GTK.java:4135)
org.eclipse.swt.widgets.Display.eventProc(Display.java:1407)
org.eclipse.swt.internal.gtk.OS._g_main_context_iteration(Native Method)
org.eclipse.swt.internal.gtk.OS.g_main_context_iteration(OS.java:1584)
org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4577)
com.biglybt.ui.swt.mainwindow.SWTThread.<init>(SWTThread.java:286)
com.biglybt.ui.swt.mainwindow.SWTThread.createInstance(SWTThread.java:56)
com.biglybt.ui.swt.Initializer.<init>(Initializer.java:179)
com.biglybt.ui.swt.UI.takeMainThread(UI.java:136)
com.biglybt.ui.Main.main(Main.java:239)
sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
java.lang.reflect.Method.invoke(Method.java:498)
com.biglybt.launcher.MainExecutor$1.run(MainExecutor.java:37)
java.lang.Thread.run(Thread.java:748)
```
</details>

<details>
<summary>GetOffSWT 1015</summary>

```
Name: GetOffSWT 1015
State: BLOCKED on java.lang.Object@5886ee1b owned by: GetOffSWT 1014
Total blocked: 750  Total waited: 1,140

Stack trace: 
com.biglybt.ui.common.table.impl.TableViewImpl._processDataSourceQueue(TableViewImpl.java:713)
com.biglybt.ui.common.table.impl.TableViewImpl.access$4(TableViewImpl.java:708)
com.biglybt.ui.common.table.impl.TableViewImpl$4.runSupport(TableViewImpl.java:913)
com.biglybt.core.util.AERunnable.run(AERunnable.java:36)
com.biglybt.core.util.ThreadPool.runIt(ThreadPool.java:328)
com.biglybt.core.util.ThreadPool$threadPoolWorker.run(ThreadPool.java:739)
com.biglybt.core.util.AEThread2$threadWrapper.run(AEThread2.java:317)
   - locked com.biglybt.core.util.AEThread2$JoinLock@3101acc9
```
</details>

<details>
<summary>GetOffSWT 1014</summary>

```
Name: GetOffSWT 1014
State: WAITING on org.eclipse.swt.internal.Lock@412edd3
Total blocked: 5  Total waited: 92

Stack trace: 
java.lang.Object.wait(Native Method)
java.lang.Object.wait(Object.java:502)
org.eclipse.swt.internal.Lock.lock(Lock.java:37)
org.eclipse.swt.internal.cairo.Cairo.cairo_surface_destroy(Cairo.java:1101)
org.eclipse.swt.graphics.Image.destroy(Image.java:928)
org.eclipse.swt.graphics.Resource.dispose(Resource.java:69)
com.biglybt.ui.swt.views.tableitems.files.ProgressGraphItem.dispose(ProgressGraphItem.java:86)
com.biglybt.ui.common.table.impl.TableColumnImpl.invokeCellDisposeListeners(TableColumnImpl.java:939)
com.biglybt.ui.swt.views.table.impl.TableCellSWTBase.dispose(TableCellSWTBase.java:594)
com.biglybt.ui.swt.views.table.impl.TableRowSWTBase.delete(TableRowSWTBase.java:177)
   - locked java.lang.Object@5886ee1b
com.biglybt.ui.swt.views.table.painted.TableRowPainted.delete(TableRowPainted.java:949)
com.biglybt.ui.common.table.impl.TableViewImpl.reallyRemoveDataSources(TableViewImpl.java:1459)
com.biglybt.ui.common.table.impl.TableViewImpl._processDataSourceQueue(TableViewImpl.java:751)
com.biglybt.ui.common.table.impl.TableViewImpl.access$4(TableViewImpl.java:708)
com.biglybt.ui.common.table.impl.TableViewImpl$4.runSupport(TableViewImpl.java:913)
com.biglybt.core.util.AERunnable.run(AERunnable.java:36)
com.biglybt.core.util.ThreadPool.runIt(ThreadPool.java:328)
com.biglybt.core.util.ThreadPool$threadPoolWorker.run(ThreadPool.java:739)
com.biglybt.core.util.AEThread2$threadWrapper.run(AEThread2.java:317)
   - locked com.biglybt.core.util.AEThread2$JoinLock@2fbb7b04
```
</details>

Is there anything to that? Or am I just jumping at shadows? The UI is definitely hung, when this happens — it'll sit there, unresponsive, forever (with torrent operations still going on in the background), until I finally hit the window manager Close button, at which point it'll wait 5 seconds and then pop up an application-not-responding dialog to let me kill it.

#### "System" Details:
```
Java 1.8.0_222 (64 bit)
  Oracle Corporation
/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.222.b10-0.fc30.x86_64/jre

SWT v4924, gtk/3.24.10, zoom=100, dpi=96
Linux v5.2.9-200.fc30.x86_64, amd64 (64 bit)
B2.0.0.1_B28/4 az3
```

**Comment by TuxPaper** [comment]
Next build should have it fixed.  If you are bored before that, you could always remove those columns and watch it not thread lock, but I mean.. you'd have to be pretty bored. :)

**Comment by ferdnyc** [comment]
I'll be honest, there have been days when you'd have caught me at least that bored... but alas, this is not one of those days.

(You know, I bet there's some clever way the progress graphs could be implemented using Unicode glyphs, so they don't even use any images. But... you'd have to be even **more** bored.))

**Comment by ferdnyc** [comment]
**NO MORE FREEZES! \o/**

And I tried _really_ hard. I even checked the "Tree view" box, _while_ "Hide files not selected..." was in the middle of building the table, _and_ I had a matching string entered into the filter field! I would've been scared to attempt even _one_ of those things on any torrent with over a couple hundred files, with the previous versions. :laughing:  Now it just takes whatever I throw at it. It's like night and day!

Thanks @TuxPaper ! Glad I finally got frustrated enough to file what _felt_, at the time, like a really half-assed, "much ado about nothing" sort of bug report. :+1:
